### PR TITLE
app/pluginservice: move plugin out from the Channels product

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
The plugin envronment and related code was tightly coupled with `Channels` product. This was causing several issues in the service initialization order. After this one gets in, we are going to decouple product hooks from the plugin environment. After that, the promise is to be able to prepare all of the services provided by `Channels` to be ready before it starts.

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: app/integration_action.go [hunk 1]
```diff
@@ -375,10 +375,10 @@ func (w *LocalResponseWriter) WriteHeader(statusCode int) {
 }
 
 func (a *App) doPluginRequest(c *request.Context, method, rawURL string, values url.Values, body []byte) (*http.Response, *model.AppError) {
-	return a.ch.doPluginRequest(c, method, rawURL, values, body)
+	return a.ch.srv.pluginService.doPluginRequest(c, method, rawURL, values, body)
```
Comment (iomodo) [inline app/integration_action.go:378]: why not pass the pluginService to the `App`? The same way we have `ch`? Law of Demeter comes to mind

**Review by iomodo — APPROVED** [review]
LGTM! One minor question below

File: app/integration_action.go (hunk 1 above)
Comment (isacikgoz) [reply app/integration_action.go:378]: This is a great point, thanks @iomodo. This is probably going to be our next step which would have an `App` struct with products. At this point, by the lifecycle, we are treating plugins like a product which makes sense.

**Review by saturninoabril — APPROVED** [review]
Thanks @isacikgoz! Tested and passed. Also, verified E2E and all good.

File: app/server.go [hunk 2]
```diff
@@ -746,6 +747,8 @@ func (s *Server) Shutdown() {
 		}
 	}
 
+	s.pluginService.ShutDownPlugins()
```
Comment (wiggin77) [inline app/server.go:750]: I think plugins should be shutdown before products.  A product won't depend on a plugin, but plugin will depend on products.

File: app/server.go [hunk 3]
```diff
@@ -848,6 +851,11 @@ func (s *Server) Start() error {
 	if err := s.products["channels"].Start(); err != nil {
 		return errors.Wrap(err, "Unable to start channels")
 	}
+
+	if err := s.InitializePluginService(); err != nil {
```
Comment (wiggin77) [inline app/server.go:855]: I think this should happen after products are started.

File: app/server.go (hunk 2 above)
Comment (isacikgoz) [reply app/server.go:750]: Makes sense 👍

File: app/server.go (hunk 3 above)
Comment (isacikgoz) [reply app/server.go:855]: Agreed, it can done once we decouple the product hooks from plugin environment.

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

**Pull request merged**